src/Catel.Core/Configuration/ConfigurationService.cs

```diff
@@ -204,18 +197,17 @@ public virtual void SetValue(ConfigurationContainer container, string key, objec
             var originalKey = key;
             key = GetFinalKey(key);
 
-            var stringValue = _objectConverterService.ConvertFromObjectToString(value, CultureInfo.InvariantCulture);
-            var existingValue = string.Empty;
+            object? existingValue;
 
             var lockObject = GetLockObject(container);
             using (lockObject.Lock())
             {
                 existingValue = GetValueFromStore(container, key);
 
-                SetValueToStore(container, key, stringValue);
+                SetValueToStore(container, key, value);
             }
 
-            if (!string.Equals(stringValue, existingValue))
+            if (!Equals(value, existingValue))
```

Comment:
I think this could even be ObjectHelper.Equals (since that does a few more things that just using the equality comparer)?

Reply:
Visual Studio tells me when adding ObjectHelper.Equals that it is redundant. So I guess it was using it already. Do you still want it written out explicitly here?

Reply:
I meant ObjectHelper.AreEqual. I am currently adding this PR to the 6.0.4 hotfix.